types/appboy-web-sdk/index.d.ts

```diff
@@ -0,0 +1,371 @@
+    localization: string;
+    manageServiceWorkerExternally: boolean;
+    minimumIntervalBetweenTriggerActionsInSeconds: number;
+    noCookies: boolean;
+    openInAppMessagesInNewTab: boolean;
+    openCardsInNewTab: boolean;
+    requireExplicitInAppMessageDismissal: boolean;
+    safariWebsitePushId: string;
+    serviceWorkerLocation: string;
+    sessionTimeoutInSeconds: number;
+}
+
+export interface AppBoy {
+    DeviceProperties: DeviceProperties;
+    changeUser(userId: string): void;
+    destroy(): void;
+    getCachedContentCards(): ContentCards;
+    getCachedFeed(): Feed;
+    getDeviceId(callback: (deviceId: string) => void): void;
+    getUser(): User;
+    initialize(apiKey: string, options?: Partial<Options>): boolean;
+    isPushBlocked(): boolean;
+    isPushGranted(yesCallback: () => void, noCallback: () => void): void; // DEPRECATED; use isPushPermissionGranted instead.
+    isPushPermissionGranted(): boolean;
+    isPushSupported(): boolean;
+    logCardClick(card: Card, forContentCards?: boolean): boolean;
+    logCardDismissal(card: Card): boolean;
+    logCardImpressions(contentCards: Card[], forContentCards?: boolean): boolean;
+    logContentCardsDisplayed(): void;
+    logCustomEvent(eventName: string, eventProperties?: object): boolean;
+    logFeedDisplayed(): void;
+    logInAppMessageButtonClick(button: Button, inAppMessage: InAppMessage): boolean;
+    logInAppMessageClick(inAppMessage: InAppMessage): boolean;
+    logInAppMessageHtmlClick(inAppMessage: HtmlMessage, buttonId?: string, url?: string): boolean;
+    logInAppMessageImpression(inAppMessage: InAppMessage|ControlMessage): boolean;
+    logPurchase(productId: string, price: number, currencyCode?: string, quantity?: number, purchaseProperties?: object): boolean;
+    openSession(): void;
+    registerAppboyPushMessages(successCallback?: (endpoint: string, publicKey: string, userAuth: string) => void, deniedCallback?: (isTemporary: boolean) => void): void;
+    removeAllSubscriptions(): void;
+    removeSubscription(subscriptionGuid: string): void;
+    requestContentCardsRefresh(): void;
+    requestFeedRefresh(): void;
+    requestImmediateDataFlush(): void;
+    resumeWebTracking(): void;
+    setLogger(loggerFunction: (message?: string) => void): void;
+    stopWebTracking(): void;
+    subscribeToContentCardsUpdates(subscriber: (contentCards: ContentCards) => void): string;
+    subscribeToFeedUpdates(subscriber: (feed: Feed) => void): string;
+    subscribeToInAppMessage(callback: (inAppMesssage: InAppMessage|ControlMessage) => void): string;
+    subscribeToNewInAppMessages(subscriber: (inAppMessages: Array<InAppMessage|ControlMessage>) => Array<InAppMessage|ControlMessage>): string; // DEPRECATED; use subscribeToInAppMessage instead.
+    toggleAppboyLogging(): void;
+    trackLocation(): void;
+    unregisterAppboyPushMessages(successCallback?: () => void, errorCallback?: () => void): void;
+    wipeData(): void;
+}
+
+export default AppBoy;
```

Comment:
I didn't find any place were a default was exported. You might know more about the library as I don't use it. If you feel like this is the correct implementation I'll trust your criteria.

Reply:
Thanks, @armanio123. I'll double check